File: app/src/main/java/com/github/overpass/gather/screen/splash/SplashViewModel.java
package com.github.overpass.gather.screen.splash;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.github.overpass.gather.model.commons.Runners;
import com.github.overpass.gather.model.data.validator.BaseValidator;
import com.github.overpass.gather.model.repo.pref.PreferenceRepo;
import com.github.overpass.gather.model.repo.register.SignUpRepo;
import com.github.overpass.gather.model.usecase.register.SignUpUseCase;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

public class SplashViewModel extends AndroidViewModel {

    private final SignUpUseCase signUpUseCase;

    public SplashViewModel(@NonNull Application application) {
        super(application);
        signUpUseCase = new SignUpUseCase(
                new SignUpRepo(FirebaseAuth.getInstance(), FirebaseFirestore.getInstance()),
                new BaseValidator(),
                new PreferenceRepo(application)
        );
    }

    public LiveData<StartStatus> onSplashAnimationComplete() {
        MutableLiveData<StartStatus> startStatusData = new MutableLiveData<>();
        Runners.io().execute(() -> {
            StartStatus startStatus = signUpUseCase.getStartStatus();
            startStatusData.postValue(startStatus);
        });
        return startStatusData;
    }
}

File: app/src/main/java/com/github/overpass/gather/screen/splash/SplashViewModel.kt
package com.github.overpass.gather.screen.splash

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.github.overpass.gather.model.commons.SingleLiveEvent
import com.github.overpass.gather.model.data.entity.splash.StartStatus
import com.github.overpass.gather.model.repo.pref.PreferenceRepo
import com.github.overpass.gather.model.repo.register.SignUpRepo
import com.github.overpass.gather.model.usecase.login.StartStatusUseCase
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore

class SplashViewModel(application: Application) : AndroidViewModel(application) {

    private val startStatusUseCase: StartStatusUseCase = StartStatusUseCase(
            PreferenceRepo(application),
            SignUpRepo(FirebaseAuth.getInstance(), FirebaseFirestore.getInstance())
    )
    private val authorizedData: SingleLiveEvent<Void> = SingleLiveEvent()
    private val unauthorizedData: SingleLiveEvent<Void> = SingleLiveEvent()
    private val notAddedDataData: SingleLiveEvent<Void> = SingleLiveEvent()
    private val unconfirmedEmailData: SingleLiveEvent<Void> = SingleLiveEvent()

    fun onSplashAnimationComplete() {
        when (startStatusUseCase.getStartStatus()) {
            StartStatus.AUTHORIZED -> authorizedData.call()
            StartStatus.UNAUTHORIZED -> unauthorizedData.call()
            StartStatus.NOT_ADDED_DATA -> notAddedDataData.call()
            StartStatus.UNCONFIRMED_EMAIL -> unconfirmedEmailData.call()
        }
    }

    fun authorized(): LiveData<Void> = authorizedData

    fun unauthorized(): LiveData<Void> = unauthorizedData

    fun notAddedData(): LiveData<Void> = notAddedDataData

    fun unconfirmedEmail(): LiveData<Void> = unconfirmedEmailData
}
